File: src/main/java/ru/atsutsiev/sd/refactoring/database/product/Product.java
package ru.atsutsiev.sd.refactoring.database.product;

import java.util.Objects;

/**
 * @author atsutsiev
 */
public class Product {
    private final String name;
    private final long price;

    public Product(final String name, final long price) {
        this.name = name;
        this.price = price;
    }

    public String getName() { return name; }

    public long getPrice() { return price; }

    public String toHTML() { return String.format("%s\t%d</br>", name, price); }

    @Override
    public String toString() { return String.format("Product {  name: %s price: %d}", name, price); }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Product product = (Product) o;
        return price == product.price && Objects.equals(name, product.name);
    }

    @Override
    public int hashCode() { return Objects.hash(name, price); }
}

File: src/main/kotlin/ru/atsutsiev/sd/refactoring/database/product/Product.kt
package ru.atsutsiev.sd.refactoring.database.product

import java.util.*

/**
 * @author atsutsiev
 */
class Product(val name: String, val price: Long) {
    fun toHTML(): String = "$name\t$price</br>"

    override fun toString(): String = "Product {  name: $name price: $price}"

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val product = other as Product
        return price == product.price && name == product.name
    }

    override fun hashCode(): Int = Objects.hash(name, price)
}